fog/sig/authority/src/ristretto.rs

```diff
@@ -0,0 +1,59 @@
+// Copyright 2018-2021 The MobileCoin Foundation
+
+//! This module contains the implementation of the fog authority signing and
+//! verification services for ristretto keys.
+//
+// TODO: If/when we have a ViewPublic/ViewPrivate types, this should be
+//       implemented on those types, instead of all ristretto public keys.
+
+use crate::{Signer, Verifier};
+use alloc::{format, string::String};
+use mc_crypto_keys::{RistrettoPrivate, RistrettoPublic, RistrettoSignature};
+
+impl Signer for RistrettoPrivate {
+    type Sig = RistrettoSignature;
+    type Error = String;
+
+    fn sign_authority(&self, spki_bytes: &[u8]) -> Result<Self::Sig, String> {
+        Ok(self.sign_schnorrkel(super::context(), spki_bytes))
+    }
```

Comment:
Does this function need to return result? it looks like it is infallible, and it looks like the caller is unwrapping the error right now

Reply:
The `signature::Signer::sign()` crate allows fallible signatures, because it's designed to work with HSMs (where you can fail to contact the HSM, because, e.g. the thing isn't plugged in). I'm carrying that through for all this signature stuff to allow for more secure key storage options in the future.